virtcontainers/acrn_arch_base.go

```diff
@@ -0,0 +1,762 @@
+		kernelParams:         acrnKernelParams,
+	}
+
+	a.handleImagePath(config)
+	return a
+}
+
+func (a *acrnArchBase) acrnPath() (string, error) {
+	p := a.path
+	return p, nil
+}
+
+func (a *acrnArchBase) acrnctlPath() (string, error) {
+	ctlpath := a.ctlpath
+	return ctlpath, nil
+}
+
+func (a *acrnArchBase) kernelParameters(debug bool) []Param {
+	params := a.kernelParams
+
+	if debug {
+		params = append(params, a.kernelParamsDebug...)
+	} else {
+		params = append(params, a.kernelParamsNonDebug...)
+	}
+
+	return params
+}
+
+func (a *acrnArchBase) memoryTopology(memoryMb uint64) Memory {
+
+	mem := fmt.Sprintf("%dM", memoryMb)
+	memory := Memory{
+		Size: mem,
+	}
+
+	return memory
+}
+
+func (a *acrnArchBase) capabilities() types.Capabilities {
+	var caps types.Capabilities
+
+	// For devicemapper disable support for filesystem sharing
+
+	caps.SetFsSharingUnsupported()
+	caps.SetBlockDeviceSupport()
+	caps.SetBlockDeviceHotplugSupport()
+
+	return caps
+}
+
+// Valid returns true if the CharDevice structure is valid and complete.
+func (cdev ConsoleDevice) Valid() bool {
+
+	if cdev.Backend != "tty" && cdev.Backend != "pty" &&
+		cdev.Backend != "console" && cdev.Backend != "socket" &&
+		cdev.Backend != "file" {
+		return false
+	}
+
+	return true
+}
+
+// AcrnParams returns the acrn parameters built out of this console device.
+func (cdev ConsoleDevice) AcrnParams(slot int, config *Config) []string {
+
+	var acrnParams []string
+	var deviceParams []string
+
+	acrnParams = append(acrnParams, "-s")
+	deviceParams = append(deviceParams, fmt.Sprintf("%d,virtio-console,", slot))
+
+	if cdev.PortType == ConsoleBE {
+		deviceParams = append(deviceParams, "@")
+	}
+
+	switch cdev.Backend {
+	case "pty":
+		deviceParams = append(deviceParams, "pty:pty_port")
+	case "tty":
+		deviceParams = append(deviceParams, fmt.Sprintf("tty:tty_port=%s", cdev.Path))
+	case "socket":
+		deviceParams = append(deviceParams, fmt.Sprintf("socket:%s=%s", cdev.Name, cdev.Path))
+	case "file":
+		deviceParams = append(deviceParams, fmt.Sprintf("file:file_port=%s", cdev.Path))
+	case "stdio":
+		deviceParams = append(deviceParams, "stdio:stdio_port")
+	default:
+		// do nothing. Error should be already caught
+	}
+
+	acrnParams = append(acrnParams, strings.Join(deviceParams, ""))
+	return acrnParams
+}
+
+// AcrnNetdevParam converts to the ACRN type to string
+func (netdev NetDevice) AcrnNetdevParam() []string {
+	var deviceParams []string
+
+	switch netdev.Type {
+	case TAP:
+		deviceParams = append(deviceParams, netdev.IFName)
+		deviceParams = append(deviceParams, fmt.Sprintf(",mac=%s", netdev.MACAddress))
+	case MACVTAP:
+		deviceParams = append(deviceParams, netdev.IFName)
+	default:
+		deviceParams = append(deviceParams, netdev.IFName)
+
+	}
+
+	return deviceParams
+}
+
+// Valid returns true if the NetDevice structure is valid and complete.
+func (netdev NetDevice) Valid() bool {
+	switch netdev.Type {
+	case TAP:
+		return true
+	case MACVTAP:
```

Comment:
Are macvtap devices not supported? If so, how come you are handling macvtap devices in AcrnNetdevParam func above?

Reply:
macvtap is supported. changed to true. Good catch!